### DISCUSSION TITLE
How I can make working a markdown MyST/reStructrueText for content of the need?

### DISCUSSION THREAD

**Discussion by petr-kus** [useblocks/sphinx-needs#1212] [Q&A]
I want to use the full potential of markdown in needs_extra_options and needs_options. 

But for the content of these options, MyST and even reStructuredText text don't work. 

to better understand see the example on the screenshot:
![public_example](https://github.com/user-attachments/assets/08f4c28c-a5eb-4122-a32c-e5d2a90986d5)

**Is there any good way or workaround for using it?** 

I already studied the templates for needs and I didn't find the solution there.

**Comment by petr-kus** [comment]
Note: It works only in content during definition as I would expect it should work everywhere...
![works_only_in_content](https://github.com/user-attachments/assets/183a62cc-5848-4e83-895d-2a48a7402356)

**Comment by danwos** [comment]
The need options do not support rendered content. No matter if it is rst or md code.
The `content` field always stores the original code of the content only, without any interpretation.

Options are meta-data and therefore not planned to hold such complex data.

I know there are use cases where such a restriction is not helpful, but it's nearly impossible to duplicate such rendered rst/md-code in a safe way. For instance, it could contain labels, sections, and other information, which should be unique. Or the rendered content destroys the layout of a needtable, as it contains images or tables.

So, what does/does not work:
✅ Use rst/md in an option
✅ Get a rendered representation of this option in the content area by using [templates](https://sphinx-needs.readthedocs.io/en/latest/directives/need.html#template). For instance to add comment after the normal content.
✅ Use the option still in filter strings
❌ Get a rendered representation inside the meta-area of a Need representation
❌ Get a rendered representation inside needtables and co.

**Reply by petr-kus** [reply]
I have developed a temporary workaround by creating a small JavaScript script that is applied to the need-tables. This script adds links to needs, web links, interprets MyST links, and includes citations and glossary links.

However, this workaround results in the loss of checks in the MyST tree during the build process. I would greatly appreciate it if a fix could be implemented somehow in your code to address this issue.
